src/WalletRpcPlugin.ts

```diff
@@ -0,0 +1,83 @@
+import { Web3PluginBase, utils, validator } from "web3";
+import {
+  AddEthereumChainRequest,
+  GetOwnedAssetsRequest,
+  GetOwnedAssetsResult,
+  SwitchEthereumChainRequest,
+  UpdateEthereumChainRequest,
+} from "./types";
+
+type WalletRpcApi = {
+  wallet_addEthereumChain: (param: AddEthereumChainRequest) => null;
+  wallet_updateEthereumChain: (param: UpdateEthereumChainRequest) => null;
+  wallet_switchEthereumChain: (param: SwitchEthereumChainRequest) => null;
+  wallet_getOwnedAssets: (param: GetOwnedAssetsRequest) => GetOwnedAssetsResult;
+};
+
+export class WalletRpcPlugin extends Web3PluginBase<WalletRpcApi> {
+  public pluginNamespace = "walletRpc";
+
+  public constructor() {
+    super();
+  }
+
+  public async addEthereumChain(param: AddEthereumChainRequest): Promise<null> {
```

Comment:
need to add doc comments for all methods

Reply:
done